Task018/main.py

```diff
@@ -0,0 +1,24 @@
+# Задача 18: Требуется найти в массиве A[1..N] самый близкий по величине элемент к заданному числу X.
+#  Пользователь в первой строке вводит натуральное число N – количество элементов в массиве. В последующих
+#  строках записаны N целых чисел Ai. Последняя строка содержит число X
+
+# *Пример:*
+
+# 5
+#     1 2 3 4 5
+#     6
+#     -> 5
+
+import random
+
+N = 30
+list = [random.randrange(1, 100) for _ in range(N)]
+
+print(list)
+
+X = int(input("Enter X: "))
+
+for i in range(len(list)):
```

Comment:
Вас уже есть параметр, который хранит длину списка.
```suggestion
for i in range(N):
```